testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestAvroSchemaEvolution.java

```diff
@@ -0,0 +1,285 @@
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(
+                                row("string_col", "varchar", "", ""),
+                                row("int_col", "integer", "", ""),
+                                row("int_col_added", "integer", "", "")));
+        assertThat(query(selectStarStatement))
+                .containsExactlyInOrder(
+                        createRow(0, "stringA0", 0, 100));
+
+        alterTableSchemaTo(REMOVED_COLUMN_SCHEMA);
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(
+                                row("int_col", "integer", "", "")));
+        assertThat(query(selectStarStatement))
+                .containsExactlyInOrder(
+                        createRow(0, 0));
+
+        alterTableSchemaTo(RENAMED_COLUMN_SCHEMA);
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(
+                                row("string_col", "varchar", "", ""),
+                                row("int_col_renamed", "integer", "", "")));
+
+        assertThat(query(selectStarStatement))
+                .containsExactlyInOrder(
+                        createRow(0, "stringA0", null));
+    }
+
+    @Test(groups = AVRO)
+    public void testSchemaWhenUrlIsUnset()
+    {
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(
+                                row("string_col", "varchar", "", ""),
+                                row("int_col", "integer", "", "")));
+        assertThat(query(selectStarStatement))
+                .containsExactlyInOrder(
+                        createRow(0, "stringA0", 0));
+
+        onHive().executeQuery(format("ALTER TABLE %s UNSET TBLPROPERTIES('avro.schema.url')", tableName));
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(
+                                row("dummy_col", "varchar", "", "")));
+    }
+
+    protected void alterTableSchemaTo(String schema)
+    {
+        onHive().executeQuery(format("ALTER TABLE %s SET TBLPROPERTIES('avro.schema.url'='%s')", tableName, schema));
+    }
+
+    private void insertData(String tableName, int rowNo, String... sqlValues)
+    {
+        String columnValues = Joiner.on(", ").join(sqlValues) +
+                (partitions.isEmpty() ? "" : ", " + getPartitionsAsListString(p -> "'" + p.getValueForRow(rowNo) + "'"));
+        query(format("INSERT INTO %s VALUES (%s)", tableName, columnValues));
+    }
+
+    private Row createRow(int rowNumber, Object... data)
+    {
+        List<Object> rowData = new ArrayList<>(Arrays.asList(data));
```

Comment:
nit: we could have `Row.builder()` and use it here

Reply:
yep, but this would required changing tempto library. And I think it's a rare case that we need to append data. In most cases i see we always have all data for row. So Object... data is enough in those cases.